__tests__/__snapshots__/runner.js.snap

```diff
@@ -1611,7 +1611,7 @@ export class PetshopApi {
         $queryParameters ? : any,
         $domain ? : string,
         $path ? : string | ((path: string) => string)
-    }): Promise < ResponseWithBody < void >> {
+    }): Promise < ResponseWithBody < object >> {
```

Comment:
Hm, should we attempt to keep these as void (i.e. avoid the object) to show clearer that no data is returned? Semantically that sounds cleaner

Reply:
I agree, `object` type would be incorrect. And a non-value type like `void` would be better suited.

The PetshopApi response types:
```
"responses": { "405": { "description": "Invalid input" } },
```

The spec:
> To indicate the response body is empty, do not specify a content for the response

https://swagger.io/docs/specification/describing-responses/#empty

Any type that could represent an "empty" non-value type would be a good candidate. `void`, `undefined`, `null` etc.

Also for the sake of backwards compatibility `void` is preferred.

I'll try fix this issue tomorrow.

Reply:
Apologies, I've been inactive. An update.

I had a go at implementing the changes, it needs some changes in typescript.ts convertType(...) function.

Needs an explicit void result type conversion.

```
export function convertType(
  swaggerType: SwaggerType,
  swagger: Swagger
): TypeSpec {
  if (isSchema(swaggerType)) {
    return convertType(swaggerType.schema, swagger);
  } else if (isReference(swaggerType)) {
    return makeReferenceTypeSpec(swaggerType);
  } else if (isEnum(swaggerType)) {
    return makeEnumTypeSpec(swaggerType);
  } else if (isString(swaggerType)) {
    return makeStringTypeSpec(swaggerType);
  } else if (isNumber(swaggerType)) {
    return makeNumberTypeSpec(swaggerType);
  } else if (isBoolean(swaggerType)) {
    return makeBooleanTypeSpec(swaggerType);
  } else if (isArray(swaggerType)) {
    return makeArrayTypeSpec(swaggerType, swagger);
  } else if (isDictionary(swaggerType)) {
    // case where a it's a Dictionary<string, someType>
    return makeDictionaryTypeSpec(swaggerType, swagger);
  } else if (
    swaggerType.minItems >= 0 &&
    swaggerType.hasOwnProperty("title") &&
    !swaggerType.$ref
  ) {
    return makeAnyTypeSpec(swaggerType);
  }

  // Remaining types are created as objects
  return makeObjectTypeSpec(swaggerType, swagger);
}
```

I'll have another shot, just I have limited time and effort.
And some other real life priorities.

Reply:
No worries, there is no rush, just take your time! :)

Reply:
This looks good!